File: src/main/java/uk/laxd/dndSimulator/dice/RollModifier.java
package uk.laxd.dndSimulator.dice;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;

/**
 * Represents a modification that needs to be made to a die roll, for example:
 *
 * * Advantage - Roll twice, take highest
 * * Disadvantage - Roll twice, take lowest
 * * Bless - Add a d4
 *
 */
public abstract class RollModifier {
    public void modifyDice(Roll roll) {}

    public void modifyRoll(Roll roll) {}
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/dice/RollModifier.kt
package uk.laxd.dndSimulator.dice

/**
 * Represents a modification that needs to be made to a die roll, for example:
 *
 * * Advantage - Roll twice, take highest
 * * Disadvantage - Roll twice, take lowest
 * * Bless - Add a d4
 *
 */
abstract class RollModifier {
    fun modifyDice(roll: Roll) {}
    fun modifyRoll(roll: Roll) {}
}